Makefile

```diff
@@ -82,11 +82,11 @@ migrate: ## apply database migrations
 html_coverage: ## generate and view HTML coverage report
 	coverage html && open htmlcov/index.html
 
-# Define PIP_COMPILE_OPTS=-v to get more information during make upgrade.
-PIP_COMPILE = pip-compile --upgrade $(PIP_COMPILE_OPTS)
+# Define COMPILE_OPTS=-v to get more information during make upgrade.
+PIP_COMPILE = pip-compile $(COMPILE_OPTS)
 
-upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
-upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+compile-requirements: export CUSTOM_COMPILE_COMMAND=make upgrade
+compile-requirements: ## Re-compile *.in requirements to *.txt
```

Comment:
Is this what you wanted?

Reply:
Yes -- this is what we do in edx-platform, where I copied these lines from. The idea is that `make compile-requirements` is the fundamental implementation here, and `make upgrade` becomes a wrapper around it that can apply an extra `--upgrade` argument.